### ISSUE TITLE
Lists nested within branches of a choice don't work

### ISSUE THREAD

**Issue by timmartin** [robshakir/pyangbind#51]
If I have a choice node that has a list node in each subtree (with the same name in each), then I can't insert elements into the list. I've got a branch that illustrates this in a unit test (https://github.com/timmartin/pyangbind/tree/nested_branch_list), but here's the summary:

Yang:

```
choice choice-one {
   case case-one {
      container case-one-container {
         list user {
            key "username";
            //...
         }
      }
      container case-two-container {
         list user {
            key "username";
            //...
            }
         }
      }
   }
}
```

Python:
```
>>> t.container.case_one_container.user.add("first")
>>> t.container.case_one_container.user["first"]
KeyError: 'first'
```

I notice that in the `bindings.py` there are two identically-named classes (`yc_user_choice__container_choice_one_user`) generated for the list in each branch. I haven't figured out whether this is the root cause.

**Comment by robshakir** [comment]
Hi Tim,

(Apologies, I forgot 'Fix #51' in the commit msg would close this!  Re-opening!)

Thanks for pointing this out - this was a bug with exactly what you pointed out -- the choice's name rather than the nested container's name was being used in the path which meant that there were duplicate classes produced. I tried this with `--split-class-dir` to see whether it was simply in this case, but it was more generic, and found another bug there (with ensuring that the right sub-modules are imported).

I integrated your unit tests into the `tests/choice` test, and added another in `tests/split-class`. Can you please pull `master` and check whether this is fixed for you?

Thanks for the super-useful bug report!

Best,
r.

**Comment by timmartin** [comment]
Yes, this works for me. Thanks for the quick turnaround.
